### PR TITLE
Improve ANET Application layout

### PR DESCRIPTION
**Pull request by VassilIordanov**
The overall layout of ANET has been revisited to better make use of the space at various window sizes.
ANET is now expected to behave better when shown on a phone or a tablet.

### CODE REVIEW

**Comment by VassilIordanov** [comment]
one quirk left - scrolling to overshoots a bit. What do you think? I could make the Scroll.Element be the title of the Fieldsets instead of the whole sections. Thoughts?

**Comment by VassilIordanov** [comment]
Also, tested only under FF and Chrome, so IE11 as no BrowserStack access from the office

File: client/src/components/Header.js [hunk 1]
```diff
@@ -22,24 +21,25 @@ export default class Header extends Component {
...
 					<Row>
-						<Col xs={3}>
-							{
-								this.props.minimalHeader ?
-									<span className="logo"><img src={logo} alt="ANET Logo" /></span> :
-									<Link to="/" className="logo">
-										<img src={logo} alt="ANET logo" />
-									</Link>
-							}
+						<Col xs={3} md={2} lg={2}>
```
Comment (gjvoosten) [inline client/src/components/Header.js]: For clarity, add `sm={3}` here.

File: client/src/index.css [hunk 2]
```diff
@@ -12,15 +12,11 @@ html {
 
 body {
 	background: #f2f2f2;
+	overflow:hidden;
```
Comment (gjvoosten) [inline client/src/index.css]: Please use a space after the colon in CSS rules.

File: client/src/index.css [hunk 3]
```diff
@@ -329,10 +325,21 @@ li#search-nav .nav > li a {
...
+.nav-fixed {
+	flex:0 0 auto;
+	padding:8px;
+	overflow-y:scroll;
+}
+
+.nav-overlay {
+	padding:8px;
+	border: 1px solid #e0e0e0;
+    height: 100vh;
```
Comment (gjvoosten) [inline client/src/index.css]: Indentation is off.

File: client/src/pages/organizations/Laydown.js [hunk 4]
```diff
@@ -35,7 +36,7 @@ class BaseOrganizationLaydown extends Component {
 		const positionsNeedingAttention = org.positions.filter(position => !position.person )
 		const supportedPositions = org.positions.filter(position => positionsNeedingAttention.indexOf(position) === -1)
 
-		return <div id="laydown" data-jumptarget>
```
Comment (gjvoosten) [inline client/src/pages/organizations/Laydown.js]: Also remove the `data-jumptarget` attribute from `client/src/pages/insights/Show.js`.

File: client/src/components/Header.js [hunk 5]
```diff
@@ -7,12 +7,11 @@ import CreateButton from 'components/CreateButton'
 import {Person} from 'models'
 
 import logo from 'resources/logo.png'
+import menuLogo from 'resources/anet-menu.png'
 
 const backgroundCss = {
 	background: '#fff',
 	paddingTop: '2em',
-	height: '100%',
-	boxShadow: '0 4px 3px 0 rgba(0,0,0,0.1)',
```
Comment (gjvoosten) [inline client/src/components/Header.js]: I think you should retain the drop-shadow below the header to make it stand out more.

File: client/src/index.css (hunk 3 above)
Comment (VassilIordanov) [reply client/src/index.css]: 337 had 4 whitespaces, while the other lines are tabs. in my editor it looks flush. I am changing it to a tab to be consistent. I assume the tools we are planning to use will enforce it one way or the other

**Comment by VassilIordanov** [comment]
@evdhoudt I took a look under IE11. A few minor comments:
- overall everything is much smoother and nicer, it feels much better :)
- if once creates a report, then shrinks the window, a horizontal sroll-bar appears
- the hiding/showing nav bar - it now occupies the entire left side, while it used to be over the left of the main pane, but was not covering any part of the header with the search bar. This had 2 purposes: 1. esthetic purpose (I thought it is cool if the ANET logo, search, etc remains visible) which is debatable and 2. making sure the security banner is always visible. This is a hard requirement as the classification of data should always remain visible.

**Comment by evdhoudt** [comment]
> @evdhoudt I took a look under IE11. A few minor comments:
> 
> * overall everything is much smoother and nicer, it feels much better :)
> * if once creates a report, then shrinks the window, a horizontal srollbar appears
> * the hiding/showing nav bar - it now occupies the entire left side, while it used to be over the left of the main pane, but was not covering any part of the header with the search bar. This had 2 purposes: 1. esthetic purpose (I thought it is cool if the ANET logo, search, etc remains visible) which is debatable and 2. making sure the security banner is always visible. This is a hard requirement as the classification of data should always remain visible. (I understand that doing so happens to give more real estate to the nab, especially as we do not have scrolling enabled in the nav)

Cool :) I'll fix the sidebar and look into the create report then shrink window scrollbar.

**Comment by evdhoudt** [comment]
@VassilIordanov I've committed the changes for IE11 and split up the App.js and ResponsiveLayout code as well. By splitting the concerns the code is much more readable :) 

I've tested the IE11 changes in browserstack, so if you could check if the changes are also working on a live IE11 environment, that would be great!

File: client/src/components/TopBar.js [hunk 6]
```diff
@@ -68,11 +68,13 @@ class BaseTopBar extends Component {
 
     render() {
         return (
-            <div style={{ flex:'0 0 auto'}}>
-                {this.props.currentUser && this.props.position && this.props.position.id === 0 && !this.props.isNewUser() && <NoPositionBanner />}
-                <GeneralBanner options={this.bannerOptions()} />
-                <SecurityBanner location={this.props.location} />
-                <Header minimalHeader={this.props.minimalHeader} toggleMenuAction={this.props.toggleMenuAction}/>
+            <div style={{ display: 'block', width: '100%', position: 'fixed'}}>
```
Comment (VassilIordanov) [inline client/src/components/TopBar.js]: A fixed layout breaks when the layout when the banner is bigger (i.e. when a banner announcement is on).
What is the "clicking add filters" problem being solved?

File: client/src/index.css [hunk 7]
```diff
@@ -331,12 +340,8 @@ li#search-nav .nav > li a {
 	padding-right: 4px;
 }
 
-.nav-fixed {
-	flex:0 0 auto;
-	padding:8px;
-}
-
 .nav-overlay {
+	margin-top: -125px;
```
Comment (VassilIordanov) [inline client/src/index.css]: I suggest to avoid offsets

File: client/src/components/TopBar.js (hunk 6 above)
Comment (VassilIordanov) [reply client/src/components/TopBar.js]: Ah, is that overcome by https://github.com/NCI-Agency/anet/pull/651/commits/f98a544cb7b3ee8263750b93fb2da964959f3ca9 ?

**Comment by VassilIordanov** [comment]
@gjvoosten  want to take another look? I have no additional inputs at the moment, so I suggest to merge it in candidate

**Comment by evdhoudt** [comment]
@VassilIordanov @gjvoosten The Firefox and IE bug is fixed. Looked at the branch together with @maradragan and we couldn't find any obvious errors in the layout. Could somebody else also take a look?

**Pull request merged**